hashing/ripemd_160.cpp

```diff
@@ -0,0 +1,457 @@
+                            0xffffffff);
+        word_data.push_back((static_cast<uint64_t>(data.size()) >> 29) &
+                            0xffffffff);
+
+        return word_data;
+    }
+
+    /**
+     * @brief implements f(j,x,y,z)
+     * @param j round number j / 16
+     * @param x value x
+     * @param y value y
+     * @param z value z
+     * @return returns the function value
+     */
+    uint32_t f(int j, uint32_t x, uint32_t y, uint32_t z) {
+        if (j == 0) {
+            return x ^ y ^ z;
+        }
+
+        else if (j == 1) {
+            return (x & y) | (~x & z);
+        }
+
+        else if (j == 2) {
+            return (x | ~y) ^ z;
+        }
+
+        else if (j == 3) {
+            return (x & z) | (y & ~z);
+        }
+
+        else {  /// if (j == 4)
+            return x ^ (y | ~z);
+        }
+    }
+
+    /**
+     * @brief implements K value for a given j
+     * @param j round number j / 16
+     * @return appropriate K value
+     */
+    uint32_t K(int j) {
+        if (j == 0) {
+            return static_cast<uint32_t>(0x00000000);
+        }
+
+        else if (j == 1) {
+            return static_cast<uint32_t>(0x5A827999);
+        }
+
+        else if (j == 2) {
+            return static_cast<uint32_t>(0x6ED9EBA1);
+        }
+
+        else if (j == 3) {
+            return static_cast<uint32_t>(0x8F1BBCDC);
+        }
+
+        else {  /// if (j == 4)
+            return static_cast<uint32_t>(0xA953FD4E);
+        }
+    }
+
+    /**
+     * @brief implements K' value for a given j
+     * @param j round number j / 16
+     * @return appropriate K' value
+     */
+    uint32_t K_dash(int j) {
+        if (j == 0) {
+            return 0x50A28BE6;
+        }
+
+        else if (j == 1) {
+            return 0x5C4DD124;
+        }
+
+        else if (j == 2) {
+            return 0x6D703EF3;
+        }
+
+        else if (j == 3) {
+            return 0x7A6D76E9;
+        }
+
+        else {  /// if (j == 4)
+            return 0x00000000;
+        }
+    }
+
+    /**
+     * @brief implements r value for a given j
+     * @param j round number j
+     * @return appropriate r value
+     */
+    int r(int j) {
+        static const int r_value[80] = {
+            0, 1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15,
+            7, 4,  13, 1,  10, 6,  15, 3,  12, 0, 9,  5,  2,  14, 11, 8,
+            3, 10, 14, 4,  9,  15, 8,  1,  2,  7, 0,  6,  13, 11, 5,  12,
+            1, 9,  11, 10, 0,  8,  12, 4,  13, 3, 7,  15, 14, 5,  6,  2,
+            4, 0,  5,  9,  7,  12, 2,  10, 14, 1, 3,  8,  11, 6,  15, 13};
+
+        return r_value[j];
+    }
+
+    /**
+     * @brief implements r' value for a given j
+     * @param j round number j
+     * @return appropriate r' value
+     */
+    int r_dash(int j) {
+        static const int r_dash_value[80] = {
+            5,  14, 7,  0, 9, 2,  11, 4,  13, 6,  15, 8,  1,  10, 3,  12,
```

Comment:
why are there so many functions which just return an array. why not just define these as members of the class?

Reply:
we are planning to moving to C++17 #2746

Reply:
So, shall I use features of C++17 and submit it?

Reply:
> So, shall I use features of C++17 and submit it?

yes. I will try to get the C++17 pr merged soonish